### PR TITLE
fix: redshift stack upgrade issue in UPDATE_ROLLBACK_FAILED

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)
1. add doc how to handle upgrade  issue 'UPDATE_ROLLBACK_FAILED ' from v1.0.x to v1.1.x
2. remove `updatable` flag for sql file, simplify the update/upgrade logic
3. change exec SQL one by one, not batch
4. ignore SQL error: `xxxx already exists` when apply SQL
5. add env `SUPPRESS_DB_ERROR` and `SUPPRESS_ALL_ERROR` to lambda `CreateSchemaForApplicationsFn` to handle further upgrade issue

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/analytics/private/app-schema.ts [hunk 1]
```diff
@@ -129,10 +129,15 @@ export class ApplicationSchemas extends Construct {
...
-      memorySize: 128,
+      memorySize: 256,
       reservedConcurrentExecutions: 1,
-      timeout: Duration.minutes(5),
+      timeout: Duration.minutes(15),
       logRetention: RetentionDays.ONE_WEEK,
+      environment: {
+        SUPPRESS_DB_ERROR: 'true',
+        SUPPRESS_ALL_ERROR: 'false',
+        ModifiedTime: `${this.getLatestModifyTimestamp()}`,
```
Comment (zxkane) [inline src/analytics/private/app-schema.ts:139]: Is it useful?

File: test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts [hunk 2]
```diff
@@ -353,93 +214,55 @@ describe('Custom resource - Create schemas for applications in Redshift database
     expect(redshiftDataMock).toHaveReceivedCommandTimes(DescribeStatementCommand, 2);
   });
 
-  test('Created bi user with lower case characters only', async () => {
```
Comment (zxkane) [inline test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts:356]: Is it changed by this fix?

File: test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts [hunk 3]
```diff
@@ -353,93 +214,55 @@ describe('Custom resource - Create schemas for applications in Redshift database
...
+
     expect(resp.Status).toEqual('SUCCESS');
-    expect(redshiftDataMock).toHaveReceivedCommandTimes(ExecuteStatementCommand, 2);
-    expect(redshiftDataMock).toHaveReceivedCommandTimes(BatchExecuteStatementCommand, 2);
-    expect(redshiftDataMock).toHaveReceivedNthSpecificCommandWith(1, BatchExecuteStatementCommand, {
-      Sqls: expect.arrayContaining([
-        expect.stringMatching(`GRANT USAGE ON SCHEMA app1 TO ${biUserNamePrefix}\\w{8}`),
-      ]),
-    });
+    expect(redshiftDataMock).toHaveReceivedCommandTimes(ExecuteStatementCommand, 44);
```
Comment (zxkane) [inline test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts:228]: Let's have a better way to get this number, for example, viewCount * 2 + reportingCount * 2 + biUserSQLCount + databaseSQLCount

File: docs/en/upgrade.md [hunk 4]
```diff
@@ -91,6 +91,49 @@ When you upgraded the pipeline from v1.0.x, you need to perform the below action
     DROP PROCEDURE  "<app-id>".sp_migrate_ods_events_1_0_to_1_1();
     ```
 
+## Trouble shooting
+
+### Clickstream-DataModelingRedshift-xxxxx stack upgrade from 1.0.x to 1.1.x failed in UPDATE_ROLLBACK_FAILED
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply docs/en/upgrade.md:96]: changed.

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 5]
```diff
@@ -47,6 +47,8 @@ const secretManagerClient = new SecretsManagerClient({
 
 export const physicalIdPrefix = 'create-redshift-db-schemas-custom-resource-';
 export const handler: CdkCustomResourceHandler = async (event: CloudFormationCustomResourceEvent, context: Context) => {
+  logger.info('Received event', { event });
```
[the remark this answers is not in the document]
Comment (amliuyong) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:50]: removed

File: test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts (hunk 3 above)
Comment (amliuyong) [reply test/analytics/analytics-on-redshift/lambda/custom-resources/create-schemas.test.ts:228]: chagned

File: src/analytics/private/app-schema.ts (hunk 1 above)
Comment (amliuyong) [reply src/analytics/private/app-schema.ts:139]: removed

**Review by zxkane — APPROVED** [review]

**Review by zxkane — APPROVED** [review]

**Pull request merged**